app/login/page.tsx

```diff
@@ -0,0 +1,162 @@
+'use client';
+
+import Link from 'next/link';
+import { useRouter } from 'next/navigation';
+import { useState } from 'react';
+import { login } from './login';
+import { isAuth } from '@/app/lib/isAuth';
+import Navigation from '../ui/Navigation';
+
+const Login = () => {
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
+  const [error, setError] = useState('');
+  const [emailError, setEmailError] = useState('');
+  const [passwordError, setPasswordError] = useState('');
+  const router = useRouter();
+
+  const validateEmail = (email: string) => {
+    const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+    return emailRegex.test(email);
+  };
+
+  const validatePassword = (password: string) => {
+    const passwordRegex =
+      /^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[!@#$%^&*()_+\-=[\]{}|;:'",.<>?])[a-zA-Z\d!@#$%^&*()_+\-=[\]{}|;:'",.<>?]{8,}$/;
+    return passwordRegex.test(password);
+  };
+
+  const handleSubmit = async (e: React.FormEvent) => {
+    e.preventDefault();
+    setError('');
+    setEmailError('');
+    setPasswordError('');
+
+    let valid = true;
+
+    if (!validateEmail(email)) {
+      setEmailError('Invalid email address');
+      valid = false;
+    }
+
+    if (!validatePassword(password)) {
+      setPasswordError(
+        'Password must contain at least 8 characters, one uppercase letter, one lowercase letter, and one number',
+      );
+      valid = false;
+    }
+
+    if (!valid) {
+      return;
+    }
+
+    try {
+      await login(email, password);
+      isAuth();
+      Navigation;
```

Comment:
Navigation is a TSX, it is a react component. You could not just put it. You should render it. So in that case just remove it. It is unnecessary here

Reply:
Removed Navigation from login page component.